### PR TITLE
sap_ha_pacemaker_cluster: ASCS ERS Simple Mount

### PR DESCRIPTION
**Pull request by marcelmamula**
@ja9fuchs Lets review it together next week when you are available.
  
This PR adds new feature: Simple Mount scenario for ASCS ERS cluster.

Changes:

1. Simple Mount added as separate task, enabled by as default ASCS cluster for SUSE, disabled for Red Hat and marked as TODO when SAPStartSrv agent is available.  Controlled by variable `sap_ha_pacemaker_cluster_nwas_abap_ascs_ers_simple_mount`
2. Added more variables for cluster resources and updated naming to align with previous changes: variables for order, colocation and groups.
3. Added IS_ERS parameter, which is valid for all 3 scenarios.
4. New entry for SAPStartSrv in resource agent dictionary `__sap_ha_pacemaker_cluster_resource_agents`

Requirement: Simple mount uses shared NFS for both ASCS and ERS data across both nodes, sap_storage_setup role has to be executed for both NFS entries on both servers by changing input in AP4S dictionary:
```yaml
      sap_storage_setup_host_type:
        - nwas_abap_ascs
        - nwas_abap_ers
```
sap_storage_setup vars in AP4S need to be adjusted as well to use actual variables from var files, instead of lookup based on hostname:
```yaml
        sap_storage_setup_nwas_abap_ascs_instance_nr: "{{ sap_system_nwas_abap_ascs_instance_nr }}"
        sap_storage_setup_nwas_abap_ers_instance_nr: "{{ sap_system_nwas_abap_ers_instance_nr }}"
```

Example of cluster with Simple Mount
```
Node List:
  * Online: [ ae1ascs ae1ers ]

Full List of Resources:
  * rsc_fence_aws       (stonith:external/ec2):  Started ae1ascs
  * Resource Group: grp_AE1_ASCS00:
    * rsc_vip_AE1_ASCS00        (ocf::heartbeat:aws-vpc-move-ip):        Started ae1ascs
    * rsc_SAPStartSrv_AE1_ASCS00        (ocf::suse:SAPStartSrv):         Started ae1ascs
    * rsc_SAPInstance_AE1_ASCS00        (ocf::heartbeat:SAPInstance):    Started ae1ascs
  * Resource Group: grp_AE1_ERS10:
    * rsc_vip_AE1_ERS10 (ocf::heartbeat:aws-vpc-move-ip):        Started ae1ers
    * rsc_SAPStartSrv_AE1_ERS10 (ocf::suse:SAPStartSrv):         Started ae1ers
    * rsc_SAPInstance_AE1_ERS10 (ocf::heartbeat:SAPInstance):    Started ae1ers
```

### CODE REVIEW

File: roles/sap_ha_pacemaker_cluster/defaults/main.yml [hunk 1]
```diff
@@ -135,6 +135,20 @@ sap_ha_pacemaker_cluster_hana_filesystem_resource_name: >-
...
   cln_SAPHanaFil_{{ sap_ha_pacemaker_cluster_hana_sid }}_HDB{{ sap_ha_pacemaker_cluster_hana_instance_nr }}
 
+# SAP HANA - Constraint names
+sap_ha_pacemaker_cluster_hana_order_topology_hana_name: >-
+  ord_saphana_saphanatop_{{ sap_ha_pacemaker_cluster_hana_sid }}_HDB{{ sap_ha_pacemaker_cluster_hana_instance_nr }}
+
+sap_ha_pacemaker_cluster_hana_colocation_hana_vip_primary_name: >-
+  col_saphana_vip_{{ sap_ha_pacemaker_cluster_hana_sid }}_HDB{{ sap_ha_pacemaker_cluster_hana_instance_nr }}_primary
+sap_ha_pacemaker_cluster_hana_colocation_hana_vip_secondary_name: >-
+  col_saphana_vip_{{ sap_ha_pacemaker_cluster_hana_sid }}_HDB{{ sap_ha_pacemaker_cluster_hana_instance_nr }}_primary
```
Comment (ja9fuchs) [inline roles/sap_ha_pacemaker_cluster/defaults/main.yml:145]: I assume this should have the `_readonly` suffix?

Comment (marcelmamula) [reply roles/sap_ha_pacemaker_cluster/defaults/main.yml:145]: @ja9fuchs Thank you, I have it fixed locally and I will push it along with crmsh post steps that I am working on.

**Comment by marcelmamula** [comment]
@ja9fuchs New post steps were added to ensure that cluster is in stable state after HAlib implementation as it was doing changes on running resources without proper restart so sapcontrol functions were not returning proper values.

It differentiates between Simple mount and classic, restarting SAPStartSrv or SAPInstance.

File: roles/sap_ha_pacemaker_cluster/tasks/Suse/post_steps_nwas_abap_ascs_ers.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    - name: "SAP HA Install Pacemaker - Put cluster in maintenance mode"
+      ansible.builtin.expect:
+        command: crm configure property maintenance-mode=true
+        responses:
+          ".*is-managed.*": "n"
+          ".*already.*": "n"
+      check_mode: false
+      changed_when: true
+
+    - name: "SAP HA Install Pacemaker - Verify that maintenace-mode is true"
```
Comment (ja9fuchs) [inline roles/sap_ha_pacemaker_cluster/tasks/Suse/post_steps_nwas_abap_ascs_ers.yml:28]: typo: maintena**n**ce-mode

File: roles/sap_ha_pacemaker_cluster/tasks/configure_nwas_ascs_ers_postinstallation.yml [hunk 3]
```diff
@@ -192,20 +263,37 @@
...
-    - name: "SAP HA Pacemaker - (SAP HA Interface) Display HA check results for ASCS"
+    - name: "SAP HA Pacemaker - (SAP HA Interface) Get HA config for ASCS"
       when:
         - __sap_ha_pacemaker_cluster_register_where_ascs.rc == 0
+      become: true
+      become_user: "{{ sap_ha_pacemaker_cluster_nwas_abap_sid | lower }}adm"
+      register: __sap_ha_pacemaker_cluster_register_ascs_ha_config
+      ansible.builtin.shell: |
+        /usr/sap/hostctrl/exe/sapcontrol -nr {{ sap_ha_pacemaker_cluster_nwas_abap_ascs_instance_nr }} -function HAGetFailoverConfig
+      changed_when: false
```
Comment (ja9fuchs) [inline roles/sap_ha_pacemaker_cluster/tasks/configure_nwas_ascs_ers_postinstallation.yml:277]: This `false` is obsolete?

File: roles/sap_ha_pacemaker_cluster/tasks/construct_vars_nwas_abap_ascs_ers.yml [hunk 4]
```diff
@@ -36,7 +36,8 @@
         value: "{{ sap_ha_pacemaker_cluster_nwas_abap_ers_sapinstance_start_profile_string }}"
       - name: AUTOMATIC_RECOVER
         value: "{{ sap_ha_pacemaker_cluster_nwas_abap_ers_sapinstance_automatic_recover_bool | string }}"
-
+      - name: IS_ERS
+        value: true
```
Comment (ja9fuchs) [inline roles/sap_ha_pacemaker_cluster/tasks/construct_vars_nwas_abap_ascs_ers.yml:40]: This is only required for ENSA1 and is part of the set_fact task that follows this ENSA2 task.
See lines 60-61.

File: roles/sap_ha_pacemaker_cluster/tasks/Suse/post_steps_nwas_abap_ascs_ers.yml (hunk 2 above)
Comment (marcelmamula) [reply roles/sap_ha_pacemaker_cluster/tasks/Suse/post_steps_nwas_abap_ascs_ers.yml:28]: Thank you, fixed in upcoming commit.

File: roles/sap_ha_pacemaker_cluster/tasks/configure_nwas_ascs_ers_postinstallation.yml (hunk 3 above)
Comment (marcelmamula) [reply roles/sap_ha_pacemaker_cluster/tasks/configure_nwas_ascs_ers_postinstallation.yml:277]: Strange that it was removed. I will add it again and retest.

File: roles/sap_ha_pacemaker_cluster/tasks/construct_vars_nwas_abap_ascs_ers.yml (hunk 4 above)
Comment (marcelmamula) [reply roles/sap_ha_pacemaker_cluster/tasks/construct_vars_nwas_abap_ascs_ers.yml:40]: Yes and No, it is still recommended configuration for ENSA2 as well, plus it is documented in our documentation for Simple mount.
https://documentation.suse.com/sbp/sap-15/html/SAP-S4HA10-setupguide-simplemount-sle15/index.html#id-implementing-the-cluster > 5.3.4 Configuring resources for the ERS Instance

Let me know and I can try to do os_family conditional as we need to keep it for Suse. @ja9fuchs

Comment (ja9fuchs) [reply roles/sap_ha_pacemaker_cluster/tasks/construct_vars_nwas_abap_ascs_ers.yml:40]: It's optional and works with both. We can keep it, our documentation does not conflict the setting. :+1:

File: roles/sap_ha_pacemaker_cluster/tasks/construct_vars_nwas_abap_ascs_ers.yml [hunk 5]
```diff
@@ -319,24 +307,15 @@
...
-# Constraints
-#################################################
-
-# Constraint parameters are pre-defined from potentially inherited ha_cluster LSR definitions.
-# Constraint definitions are combined into these parameters.
-# See tasks/ascertain_ha_cluster_in_inventory.yml:
-#
-# __sap_ha_pacemaker_cluster_constraints_colocation: "{{ ha_cluster_constraints_colocation }}"
-# __sap_ha_pacemaker_cluster_constraints_location: "{{ ha_cluster_constraints_location }}"
-# __sap_ha_pacemaker_cluster_constraints_order: "{{ ha_cluster_constraints_order }}"
```
[the remark this answers is not in the document]
Comment (ja9fuchs) [reply roles/sap_ha_pacemaker_cluster/tasks/construct_vars_nwas_abap_ascs_ers.yml:332]: True, I agree it's more consistent to remove it.

**Review by ja9fuchs — APPROVED** [review]
tested and looking good with the improved HA validation :+1:

**Pull request merged**